### PR TITLE
More doctests for some reducedim methods

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/reducedim.jl [hunk 1]
```diff
@@ -304,6 +304,22 @@ sum(A, dims)
...
 Sum elements of `A` over the singleton dimensions of `r`, and write results to `r`.
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+ 
+julia> sum!([1; 1], A)
+2-element Array{Int64,1}:
```
Comment (KristofferC) [inline base/reducedim.jl]: Would something like:

```
julia> r = [1;1]; sum!(r, A); r
2-element Array{Int64,1}:
 3
 7
```

to show that `r` get modified be useful? Perhaps it is just cluttering things.

Comment (KristofferC) [inline base/reducedim.jl]: Would something like:

```julia
julia> r = [1;1]; sum!(r, A); r
2-element Array{Int64,1}:
 3
 7
```

to show that `r` get modified be useful? Perhaps it is just cluttering things.

File: base/reducedim.jl [hunk 2]
```diff
@@ -343,13 +375,45 @@ prod!(r, A)
...
 
 Compute the maximum value of `A` over the singleton dimensions of `r`, and write results to `r`.
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> maximum!([1; 1], A)
```
Comment (KristofferC) [inline base/reducedim.jl]: Not related to the PR but this is confusing to me:

```jl
julia> maximum!([3;1], A)
2-element Array{Int64,1}:
 2
 4

julia> maximum([3;1], A)
1-element Array{Int64,1}:
 3
```

**Review by KristofferC — APPROVED** [review]

File: base/reducedim.jl (hunk 2 above)
Comment (tkelman) [inline base/reducedim.jl]: using a matrix for the dimensions argument is a little funny and could maybe be an error?

(or a keyword could be used for dims arguments but that would be a big change)

File: base/reducedim.jl [hunk 3]
```diff
@@ -510,6 +638,25 @@ end
...
+
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> findmin(A, 1)
+(
+[1 2],
```
Comment (tkelman) [inline base/reducedim.jl]: the extra lines here are odd, where are those coming from?

Comment (kshyatt) [inline base/reducedim.jl]: I agree they are odd, not sure where they're coming from.

Comment (KristofferC) [inline base/reducedim.jl]: I know it has been mentioned somewhere...

Comment (KristofferC) [inline base/reducedim.jl]: It got changed now anyway:

Currently:
```
julia> findmin(A, 1)
([0.054146 0.121874 … 0.0024117 0.0398365], [6 15 … 84 97])
```

**Comment by kshyatt** [comment]
Building this locally succeeded for me, going to merge.

**Pull request merged**